### PR TITLE
MNT: Compat with pytest>=8

### PR DESCRIPTION
**Pull request by pllim**
Because the hook check has changed in 8.1 . See https://github.com/pytest-dev/pytest/issues/11779#issuecomment-1879387034

### CODE REVIEW

File: pytest_doctestplus/plugin.py [hunk 1]
```diff
@@ -519,159 +519,299 @@ def parse(self, s, name=None):
...
+
+        def pytest_ignore_collect(self, path, config):
+            """
+            Skip paths that match any of the doctest_norecursedirs patterns or
+            if doctest_only is True then skip all regular test files (eg test_*.py).
+            """
+            if PYTEST_GE_8_0:
+                dirpath = Path(path).parent
+                collect_ignore = config._getconftest_pathlist("collect_ignore",
+                                                              path=dirpath)
```
Comment (bsipocz) [inline pytest_doctestplus/plugin.py:686]: no need for this conditional

File: pytest_doctestplus/plugin.py [hunk 2]
```diff
@@ -519,159 +519,299 @@ def parse(self, s, name=None):
...
-        Skip paths that match any of the doctest_norecursedirs patterns or
-        if doctest_only is True then skip all regular test files (eg test_*.py).
-        """
-        if PYTEST_GE_8_0:
-            dirpath = Path(path).parent
-            collect_ignore = config._getconftest_pathlist("collect_ignore",
-                                                          path=dirpath)
-        elif PYTEST_GE_7_0:
-            dirpath = Path(path).parent
+if PYTEST_GE_8_0:
```
Comment (bsipocz) [inline pytest_doctestplus/plugin.py:522]: Could this conditional be in and around methods that's affected rather than for the whole class?

**Review by bsipocz — COMMENTED** [review]
I find having the conditional around the class makes things less readable, but if you say it's better this way, I'm OK with it.

That's being said, in the else branch there are some cleanups to be made for superfluous conditional.

**Comment by bsipocz** [comment]
Oh, and please mention it in the changelog.

File: pytest_doctestplus/plugin.py (hunk 2 above)
Comment (pllim) [reply pytest_doctestplus/plugin.py:522]: I don't think so. If I understood https://github.com/pytest-dev/pytest/issues/11779#issuecomment-1879387034 correctly, I have to update the function signature, not just the internals.

File: pytest_doctestplus/plugin.py (hunk 1 above)
Comment (pllim) [reply pytest_doctestplus/plugin.py:686]: Good catch! Accidentally left this in from an earlier attempt.

File: pytest_doctestplus/plugin.py (hunk 2 above)
Comment (bsipocz) [reply pytest_doctestplus/plugin.py:522]: yeap, what I meant was to do the conditional outside of the affected methods, but not outside of the class.

**Comment by pllim** [comment]
@bsipocz , I think I addressed your comments. Also did some infrastructure updates that I noticed while porting stuff from here to https://github.com/astropy/pytest-filter-subpackage/pull/15 .

File: pytest_doctestplus/plugin.py (hunk 2 above)
Comment (pllim) [reply pytest_doctestplus/plugin.py:522]: Oh, now that you mention it, I think Marten did something like that for astropy.units . Lemme try.

Comment (pllim) [reply pytest_doctestplus/plugin.py]: Okay, that works though in the end, most of the class still need the version check. 🤷‍♀️

Comment (bsipocz) [reply pytest_doctestplus/plugin.py]: ahh, I'm sorry about it then, hoped it would end up a bit cleaner.

File: tox.ini [hunk 3]
```diff
@@ -28,9 +27,7 @@ deps =
     pytest73: pytest==7.3.*
     pytest74: pytest==7.4.*
```
Comment (bsipocz) [inline tox.ini:28]: I would already add a placeholder for the 8.0.x series (with a temporary `pre` option), given these changes affect that branch and currently, we don't have an idea whether all is fine, or it is only passing with 8.1.dev

**Review by bsipocz — APPROVED** [review]
The code changes look, good, but I would like to play it save and include 8.0.x in the test suite. Given it's already the evening for you, I'll just push a commit to this branch, and go ahead and merge if CI is all green.

**Comment by bsipocz** [comment]
Thanks @pllim!

**Comment by pllim** [comment]
Indeed I was AFK so thanks for wrapping this up! 🙇‍♀️

**Pull request merged**